node/src/config.rs

```diff
@@ -0,0 +1,154 @@
+}
+
+#[cfg(test)]
+mod tests {
+    use figment::Jail;
+    use miden_node_block_producer::config::{BlockProducerConfig, BlockProducerEndpoint};
+    use miden_node_rpc::config::{RpcConfig, RpcEndpoint};
+    use miden_node_store::config::{StoreConfig, StoreEndpoint};
+    use miden_node_utils::Config;
+
+    use super::{NodeTopLevelConfig, CONFIG_FILENAME};
+
+    #[test]
+    fn test_node_config() {
+        Jail::expect_with(|jail| {
+            jail.create_file(
+                CONFIG_FILENAME,
+                r#"
+                    [block_producer]
+                    store_endpoint = "http://store:8000"
+                    block_producer_endpoint = "http://block_producer:8001"
+
+                    [block_producer.endpoint]
+                    host = "127.0.0.1"
+                    port = 8080
+
+                    [rpc]
+                    store_endpoint = "http://store:8000"
+                    block_producer_endpoint = "http://block_producer:8001"
+
+                    [rpc.endpoint]
+                    host = "127.0.0.1"
+                    port = 8080
+
+                    [store]
+                    sqlite = "local.sqlite3"
+
+                    [store.endpoint]
+                    host = "127.0.0.1"
+                    port = 8080
+                "#,
+            )?;
+
+            let config: NodeTopLevelConfig = NodeTopLevelConfig::load_config(None).extract()?;
+
+            assert_eq!(
+                config,
+                NodeTopLevelConfig {
+                    block_producer: BlockProducerConfig {
+                        endpoint: BlockProducerEndpoint {
+                            host: "127.0.0.1".to_string(),
+                            port: 8080,
+                        },
+                        store_endpoint: "http://store:8000".to_string(),
+                    },
+                    rpc: RpcConfig {
+                        endpoint: RpcEndpoint {
+                            host: "127.0.0.1".to_string(),
+                            port: 8080,
+                        },
+                        store_endpoint: "http://store:8000".to_string(),
+                        block_producer_endpoint: "http://block_producer:8001".to_string(),
+                    },
+                    store: StoreConfig {
+                        endpoint: StoreEndpoint {
+                            host: "127.0.0.1".to_string(),
+                            port: 8080,
+                        },
+                        sqlite: "local.sqlite3".into(),
+                    },
+                }
+            );
+
+            Ok(())
+        });
+    }
+
+    #[test]
+    fn test_node_config_env() {
+        Jail::expect_with(|jail| {
+            // Block producer
+            // ------------------------------------------------------------------------------------
+            jail.set_env("MIDEN__BLOCK_PRODUCER__STORE_ENDPOINT", "http://store:8000");
```

Comment:
We should document somewhere how these environment variables work, whether or not they override anything in the config, etc

Reply:
> If it simplifies things, I'm fine with not having support for setting configs via environment variables. That is, we'd have 2 options: defaults or config settings that come from .toml files.

It simplifies the code but complicates deployment. We should keep the env variables.